packages/stream_chat_flutter_core/lib/src/message_list_core.dart

```diff
@@ -167,21 +167,23 @@ class MessageListCoreState extends State<MessageListCore> {
     QueryDirection direction = QueryDirection.top,
   }) {
     if (!_isThreadConversation) {
-      return _streamChannel.queryMessages(direction: direction);
+      return _streamChannel!.queryMessages(direction: direction);
     } else {
-      return _streamChannel.getReplies(widget.parentMessage!.id);
+      return _streamChannel!.getReplies(widget.parentMessage!.id);
     }
   }
 
-  var _initialized = false;
-
   @override
   void didChangeDependencies() {
-    _streamChannel = StreamChannel.of(context);
-    if (!_initialized && _isThreadConversation) {
-      _initialized = true;
-      _streamChannel.getReplies(widget.parentMessage!.id);
+    final newStreamChannel = StreamChannel.of(context);
+
+    if (newStreamChannel != _streamChannel) {
+      _streamChannel = newStreamChannel;
+      if (_isThreadConversation) {
+        _streamChannel!.getReplies(widget.parentMessage!.id);
```

Comment:
Do we want to do this in didUpdateWidget and once somewhere else while starting? Logic is the parent message ID is no longer valid for a new stream channel in the inherited widget, and they would need to change the parent message ID too, which would better be suited in didUpdate than here.